pkg/types/status/status.go

```diff
@@ -16,47 +16,113 @@ limitations under the License.
 
 package status
 
+import (
+	kapps "k8s.io/api/apps/v1"
+)
+
 type Status struct {
-	APIName       string `json:"api_name"`
-	APIID         string `json:"api_id"`
-	Code          Code   `json:"status_code"`
-	ReplicaCounts `json:"replica_counts"`
+	Ready         int32          `json:"ready" yaml:"ready"`
+	Requested     int32          `json:"requested" yaml:"requested"`
+	UpToDate      int32          `json:"up_to_date" yaml:"up_to_date"`
+	ReplicaCounts *ReplicaCounts `json:"replica_counts,omitempty" yaml:"replica_counts,omitempty"`
 }
 
-type ReplicaCounts struct {
-	Updated   SubReplicaCounts `json:"updated"`
-	Stale     SubReplicaCounts `json:"stale"`
-	Requested int32            `json:"requested"`
+type ReplicaCountType string
+
+const (
```

Comment:
Add comment which says that all counts are for the up to date spec, except for the one that isn't